python/ray/tests/test_runtime_env_packaging.py

```diff
@@ -514,6 +518,132 @@ def test_parse_gcs_uri(self, gcs_uri):
+                await download_and_unpack_package(
+                    pkg_uri=pkg_uri, base_directory=temp_dest_dir
+                )
+                mock_unzip_package.assert_called_once()
+
+    @pytest.mark.parametrize(
+        "pkg_uri",
+        [
+            "s3://my-bucket/my-zipfile.zip",
+            "s3://my-bucket/path/to/my-zipfile.zip",
+            "s3://my-zipfile.zip",
+        ],
+    )
+    @patch("ray._private.runtime_env.packaging.unzip_package", side_effect=None)
+    async def test_download_and_unpack_package_with_s3_uri(
+        self, mock_unzip_package, pkg_uri
+    ):
+        with tempfile.TemporaryDirectory() as temp_dest_dir:
+            with patch("smart_open.open", mock_open(read_data="some contents")):
+                await download_and_unpack_package(
+                    pkg_uri=pkg_uri, base_directory=temp_dest_dir
+                )
+                mock_unzip_package.assert_called_once()
+
+    @pytest.mark.parametrize(
+        "pkg_uri",
+        [
+            "gs://my-bucket/my-zipfile.zip",
+            "gs://my-bucket/path/to/my-zipfile.zip",
+            "gs://my-zipfile.zip",
+        ],
+    )
+    @patch("ray._private.runtime_env.packaging.unzip_package", side_effect=None)
+    async def test_download_and_unpack_package_with_gs_uri(
+        self, mock_unzip_package, pkg_uri
+    ):
+        with tempfile.TemporaryDirectory() as temp_dest_dir:
+            with patch("smart_open.open", mock_open(read_data="some contents")):
+                await download_and_unpack_package(
+                    pkg_uri=pkg_uri, base_directory=temp_dest_dir
+                )
+                mock_unzip_package.assert_called_once()
+
+    async def test_download_and_unpack_package_with_file_uri(self):
+        with (
+            tempfile.TemporaryDirectory() as src_dir,
+            tempfile.TemporaryDirectory() as temp_dest_dir,
+        ):
+            # Create a temporary file with a zip extension in the temporary directory
+            with tempfile.NamedTemporaryFile(suffix=".zip", dir=src_dir) as zip_file:
+                with zipfile.ZipFile(zip_file.name, "w") as zip:
+                    # Add a file to the zip file
+                    zip.writestr("file.txt", "Hello, world!")
+
+                pkg_uri = f"file://{zip_file.name}"
+
+                local_dir = await download_and_unpack_package(
+                    pkg_uri=pkg_uri, base_directory=temp_dest_dir
+                )
+
+                # Check that the file was extracted to the destination directory
+                assert (Path(local_dir) / "file.txt").exists()
+
+    @pytest.mark.parametrize(
+        "protocol",
+        [
+            Protocol.CONDA,
+            Protocol.PIP,
+        ],
+    )
+    async def test_download_and_unpack_package_with_unsupported_protocol(
+        self, protocol: Protocol
+    ):
+        # Test giving an unsupported protocol.
+        pkg_uri = f"{protocol.value}://some-package.zip"
```

Comment:
Totally agree. I think it would be good if we could check the suffix of the pkg_uri or at least modify the assertion error message a little to explicitly say that the problem is about the suffix. Let me add a suffix check.